doc/specs/#190 - Proxy Support.md

```diff
@@ -0,0 +1,89 @@
+---
+author: Flor Chacon @florelis
+created on: 2024-02-07
+last updated: 2024-02-07
+issue id: 190
+---
+
+# Proxy support
+
+For [#190](https://github.com/microsoft/winget-cli/issues/190)
+
+## Abstract
+
+This spec describes a feature to specify a proxy for winget to use when connecting to the internet.
+
+## Solution Design
+
+A new command line argument will be added to specify a proxy to use during a particular invocation of winget.
+This functionality will first need to be enabled through an admin setting, similar to local manifests or hash override.
+
+An option to set a default proxy to use on every flow will be added, but it will require administrator permissions to be set.
+
+New Group Policy will also be added for IT admins to control the use of proxies.
+The policies will be similar to those we already have for sources, so that a specific proxy can be required or only a predefined set of proxies can be allowed.
+
+Proxies will not be used for the configuration features for now.
+
+If a proxy is configured, it will be used for accessing sources and downloading installers.
+Note that it will not affect the behavior of installers themselves, so an installation may still generate traffic outside of the proxy.
+
+## UI/UX Design
+
+We will add a command line argument taking the URI to the proxy.
+A separate parameter will be available to disable the use of proxy if there is a default set.
+
+```
+> winget settings --enable ProxyCommandLineArgument
```

Comment:
Can we simplify here? Instead of having a separate command for managing the proxy, it could be built into the enable? Running the enable again would over-write the existing proxy. Thoughts @yao-msft?

`winget settings --enable DefaultProxy https://127.0.0.1:2345`
`winget settings --disable DefaultProxy`

Reply:
I think Flor added this for admin/group policy to control whether non-elevated user could specify a different proxy in command line arg in one invocation. like "winget install TestPackage --Proxy https://MyTestProxy.com". This is a bit different from specifying default proxy for all users.

By the way, currently admin settings only supports bool, it may be extended to support string or it's put into separate proxy settings.

Reply:
Ah, I see. So theres actually 2 different "admin settings" here. That makes sense

Reply:
I added a couple of lines to try and clarify the difference between "default proxy" and "standard user can set use proxy"